auth/providers/azure/azure.go

```diff
@@ -120,23 +122,34 @@ type metadataJSON struct {
 func getMetadata(aadEndpoint, tenantID string) (*metadataJSON, error) {
 	metadataURL := aadEndpoint + tenantID + "/.well-known/openid-configuration"
 
-	response, err := http.Get(metadataURL)
+	request, err := http.NewRequest("GET", metadataURL, nil)
 	if err != nil {
 		return nil, err
 	}
-	defer response.Body.Close()
 
-	if response.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("get metadata url failed with status code: %d", response.StatusCode)
-	}
+	// Copy the default HTTP client so we can set a timeout.
+	// (It uses the same transport though since the pointer gets copied)
+	httpClient := *httpclient.DefaultHTTPClient
```

Comment:
i don't think this is the right way. this will be creating new Transport and new connection pool every time

Reply:
It doesn't because the transport is a pointer / interface object, as the comment notes.